File: clients/py/src/features/wrap_type_bigint.py
from pathlib import Path
from typing import Any, TypedDict

from polywrap import (
    FsUriResolver,
    PolywrapClient,
    PolywrapClientConfigBuilder,
    SimpleFileReader,
    Uri,
)
from pydantic import BaseModel


class Obj(TypedDict):
    prop1: str


class Args(TypedDict):
    arg1: str
    obj: Obj


class TestCaseInput(BaseModel):
    args: Args


def run_test_case(input: Any) -> None:
    input_obj = TestCaseInput.parse_obj(input)

    root = Path(__file__).parent.parent.parent.parent.parent / "wraps"
    uri = Uri.from_str(f"fs/{root}/bigint-type/implementations/as")

    config = (
        PolywrapClientConfigBuilder()
        .add_resolver(FsUriResolver(SimpleFileReader()))
        .build()
    )

    client = PolywrapClient(config)

    print("Invoking method")

    response = client.invoke(
        uri=uri,
        method="method",
        args=input_obj.args,
    )

    if not response:
        raise ValueError(f"Error: {response}")

    bigint = int(response)

    print("Result:", bigint)
    print("Success!")

File: clients/js/src/features/wrap_type_bigint.ts
import { Input } from "../input";

import { PolywrapClientConfigBuilder, PolywrapClient } from "@polywrap/client-js";
import path from "path";

export async function runTestCase(input: unknown): Promise<void> {
  const inputObj = Input.expectObject<{
    args: unknown;
  }>(input);
  const args = Input.expectObject<Record<string, unknown>>(
    inputObj.args
  );

  const root = path.join(__dirname, "../../../../wraps");
  const uri = `fs/${root}/bigint-type/implementations/as`;

  const config = new PolywrapClientConfigBuilder()
    .addDefaults()
    .build();
  const client = new PolywrapClient(config);

  console.log("Invoking method");

  let response = await client.invoke<string>({
    uri,
    method: "method",
    args: args,
  });

  if (!response.ok) throw response.error;

  const bigint = BigInt(response.value);

  console.log("Result:", bigint.toString());
  console.log("Success!");
}
